### ISSUE TITLE
When using portkey as a provider, required request headers are not being set correctly

### ISSUE THREAD

**Issue by rajatvijay** [promptfoo/promptfoo#2075]
**Describe the bug**
Two issues with Portkey provider integration:
1. PORTKEY_API_KEY environment variable is not being read automatically
2. Header key formatting is incorrect for Portkey API key (generates 'x-portkey-apiKey' instead of 'x-portkey-api-key')

**To Reproduce**
Bug 1 - Environment Variable Not Read:
1. Set up PORTKEY_API_KEY in environment:
```bash
export PORTKEY_API_KEY=your_api_key
```

2. Create promptfoo config:
```yaml
prompts:
  - 'portkey://pp-task-gener-1e0c8d'

providers:
  - id: portkey:gpt-4o
    config:
      portkeyProvider: openai

tests: file://tests/sample-cases.csv
```

3. Run evaluation:
```bash
promptfoo eval
```

4. See error about missing Portkey API key

Bug 2 - Incorrect Header Formatting:
1. Create promptfoo config with explicit API key:
```yaml
prompts:
  - 'portkey://pp-task-gener-1e0c8d'

providers:
  - id: portkey:gpt-4o
    config:
      portkeyProvider: openai
      portkeyApiKey: your_api_key

tests: file://tests/sample-cases.csv
```

2. Run evaluation and inspect network requests:
```bash
promptfoo eval
```

3. Observe header is formatted as 'x-portkey-apiKey' instead of 'x-portkey-api-key'

**Expected behavior**
1. The Portkey provider should automatically read PORTKEY_API_KEY from environment variables when no API key is provided in config
2. When portkeyApiKey is provided in config, it should be transformed to 'x-portkey-api-key' in headers

**System information:**
- Promptfoo version: 0.94.5

**Additional context**
In src/providers/portkey.ts, there are two distinct issues:

1. Environment Variable Issue:
The constructor doesn't check for the PORTKEY_API_KEY environment variable. Unlike other providers, there's no fallback mechanism to read from environment variables when the config doesn't contain an API key.

2. Header Formatting Issue:
The header transformation logic has a bug in the string manipulation:
```typescript
const headerKey = key.startsWith('portkey')
  ? `x-${key.replace(/^portkey/, 'portkey-').toLowerCase()}`
  : key;
```
This transforms 'portkeyApiKey' to 'x-portkey-apiKey', preserving the camelCase. The Portkey API expects 'x-portkey-api-key' format. The toLowerCase() is applied after the initial replacement, but doesn't handle the camelCase conversion correctly.

Reference to correct header format can be found in Portkey's official documentation:
https://portkey.ai/docs/api-reference/inference-api/portkey-sdk-client#parameters (check "REST Headers" tab)

The documentation specifically shows the header should be 'x-portkey-api-key' in their API reference.

**Comment by rajatvijay** [comment]
@typpo does it look like something that needs to be prioritised? I can go ahead and create a PR.

**Comment by typpo** [comment]
Hi @rajatvijay, have a fix in #2088.  If possible, could you give it a try?

**Comment by typpo** [comment]
Released in 0.96.0!
